src/NServiceBus.Core/Transports/OutgoingMessage.cs

```diff
@@ -0,0 +1,32 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The message going out to the transport
+    /// </summary>
+    public class OutgoingMessage
+    {
+        /// <summary>
+        /// Constructs the message
+        /// </summary>
+        /// <param name="headers">The headers associated with this message</param>
+        /// <param name="body">The body of the message</param>
+        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
```

Comment:
>This is outgoing messages so it will already be in memory since the users
will have "new'ed" up the message? 

yes

> so you will already have allocated all the memory

no. since the act of serializing the message to a byte array effectively takes a, larger, copy of the message into memory. effectively taking up the memory space twice.